manifests/0000_70_cluster-network-operator_01_egr_crd.yaml

```diff
@@ -0,0 +1,128 @@
+# This file is automatically generated. DO NOT EDIT
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.4.0
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+  creationTimestamp: null
+  name: egressrouters.network.operator.openshift.io
+spec:
+  group: network.operator.openshift.io
+  names:
+    kind: EgressRouter
+    listKind: EgressRouterList
+    plural: egressrouters
+    shortNames:
+    - egr
+    singular: egressrouter
+  scope: Namespaced
+  versions:
+  - additionalPrinterColumns:
+    - jsonPath: .status.egressRouter
+      name: EgressRouters
+      type: string
+    - jsonPath: .status.status
+      name: Status
+      type: string
+    name: v1
+    schema:
+      openAPIV3Schema:
+        description: EgressRouter is a single egressrouter pod configuration object.
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: EgressRouterSpec contains the configuration for an egress
+              router.
+            properties:
+              ip:
+                description: IP configuration arguments.
+                properties:
+                  addresses:
+                    description: List of IP addresses to configure on the interface.
+                    items:
+                      description: EgressRouterAddress contains a pair of ip cidr
+                        and gateway to be configured on the router's interface
+                      properties:
+                        gateway:
+                          description: IP address of the next-hop gateway, if it cannot
+                            be automatically determined. Can be IPv4 or IPv6.
+                          type: string
+                        ip:
+                          description: IP is the address to configure on the router's
+                            interface. Can be IPv4 or IPv6.
+                          type: string
```

Comment:
If addresses allows a list, and this list may include both ipv4 and ipv6 addresses at the same time, then we need to be able to specify an object that allows both gateways rather than a string.
Also a pattern to validate would be nice:
https://github.com/openshift/cluster-network-operator/blob/f47200e8dbc052501e79192a5b7933e1195b1b91/bindata/network/openshift-sdn/001-crd.yaml#L48

Reply:
As commented on the link casey's reflecting, we'll handle such check on the controller code